src/Ext/SWType/Hooks.cpp

```diff
@@ -2,8 +2,10 @@
 
 #include <SuperClass.h>
 
-//Ares hooked from 0x6CC390 and jumped to this offset
-DEFINE_HOOK(0x6CDE40, SuperClass_Place_FireExt, 0x3)
+//Ares hooked at 0x6CC390 and jumped to 0x6CDE40
+// If a super is not handled by Ares however, we do it at the original entry point
+DEFINE_HOOK_AGAIN(0x6CC390, SuperClass_Place_FireExt, 0x6)
```

Comment:
```suggestion
bool SWLaunchHandledByAres=true;
DEFINE_HOOK(0x6CC390, SuperClass_Place_FireExt, 0x6)
{
    GET(SuperClass* const, pSuper, ECX);
	GET_STACK(CellStruct const* const, pCell, 0x4);
	SWLaunchHandledByAres=false;
	SWTypeExt::FireSuperWeaponExt(pSuper, *pCell);
	return 0;
}

DEFINE_HOOK(0x6CDE40, SuperClass_Place_FireExt, 0x4)
{
	GET(SuperClass* const, pSuper, ECX);
	GET_STACK(CellStruct const* const, pCell, 0x4);
	// GET_STACK(bool const, isPlayer, 0x8);
	if(!SWLaunchHandledByAres)
	    SWTypeExt::FireSuperWeaponExt(pSuper, *pCell);
	return 0;
}
```
I was thinking about this, otherwise you'll fire your ext twice, no?

Reply:
SW is fired once (tested with all types), there is no need for an additional check. 
As for hook size, there's not much choice, as it's followed by a byte of padding and a jump table.

Reply:
If it doesn't pose any problem then it should be fine, perhaps